crates/rolldown/src/ecmascript/format/umd.rs

```diff
@@ -127,6 +128,38 @@ pub fn render_umd<'code>(
     }
   });
 
+  if let ChunkKind::EntryPoint { module: entry_id, .. } = ctx.chunk.kind {
+    let entry_meta = &ctx.link_output.metas[entry_id];
+    match entry_meta.wrap_kind {
+      WrapKind::Esm => {
+        let wrapper_ref = entry_meta.wrapper_ref.as_ref().unwrap();
+        // init_xxx
+        let wrapper_ref_name = ctx.finalized_string_pattern_for_symbol_ref(
+          *wrapper_ref,
+          ctx.chunk_idx,
+          &ctx.chunk.canonical_names,
+        );
+        ctx.link_output.symbol_db.canonical_name_for(*wrapper_ref, &ctx.chunk.canonical_names);
+        source_joiner.append_source(concat_string!(wrapper_ref_name, "();"));
+      }
+      WrapKind::Cjs => {
+        let wrapper_ref = entry_meta.wrapper_ref.as_ref().unwrap();
+
+        // require_xxx
+        let wrapper_ref_name = ctx.finalized_string_pattern_for_symbol_ref(
+          *wrapper_ref,
+          ctx.chunk_idx,
+          &ctx.chunk.canonical_names,
+        );
+
+        // require_xxx();
+        // FIXME: should set the umd's exports with `require_xxx()`
+        source_joiner.append_source(concat_string!(wrapper_ref_name, "();\n"));
+      }
+      WrapKind::None => {}
+    }
+  }
+
   //  exports
   if let Some(exports) = render_chunk_exports(ctx, Some(&export_mode)) {
```

Comment:
This should be refactored with considering module being wrapped.

@underfin do you have time for this?

Reply:
iife format should be refactored too I guess.

Reply:
The logic of `module being wrapped` moving into `render_chunk_exports`?

Reply:
I guess so. Need to render different `module.exports = require_xxx()`/`return require_xxx()` based on output format.

Reply:
Yeah, you are right, but it needs to low priority for me at now, let us delay to do it.